imgui-knobs.cpp

```diff
@@ -133,6 +148,16 @@ namespace ImGuiKnobs {
                 ImGuiKnobFlags flags,
                 float angle_min,
                 float angle_max) {
+            if (flags & ImGuiKnobFlags_Logarithmic && v_min <= 0.0 && v_max >= 0.0) {
+                // we must handle the cornercase if a client specifies a logarithmic range that contains zero
```

Comment:
There's an issue that, based on this comment, seems related to this logic. Using min=-20000, max=100. It seems to use 0.1 as the min and clamp the value to it.

Reply:
Can you please explain in more detail what exactly is the issue? 

Imho the "expected" behavior is that the range is clamped to [0.1, 100.0] in that case assuming a format string of "0.1f".

Since log(0) is undefined a range containing a zero is ill-formed in the first place, to avoid issues we only use the positive part of the range. We cannot start at 0 so we need to find a reasonable minimal value. This was done using the format string as a hint inspired by ImGui::SliderBehaviorT (refer to https://github.com/ocornut/imgui/blob/7219fa65c0212de1d352b5db8f7dfe43b11f00c9/imgui_widgets.cpp#L3008)

I tried this example on my machine with above parameters and the result was range [0.1, 100.0]

https://github.com/user-attachments/assets/5489a733-f48a-4f3b-8186-0d52702d196d

Reply:
No, I disagree, the logarithm of zero is mathematically undefined and approaches negative infinity. I don't think it should be possible to specify such a range as it does not make any sense to use a logarithmic knob for a range that contains zero. The code for the catching the corner case is there to prevent the user from specifying an invalid range. My first approach was to throw an exception in this case but I then decided that potentially crashing the users application for it might be too "harsch" therefore I implemented it similar as it is implemented in the upstream ImGui project.

Reply:
Fair enough, I think that's a valid point. Reading up on it confirms your point, and I don't think we should support it with some semi-hacky solution. Thank you for explaining! :)